akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala

```diff
@@ -129,12 +129,11 @@ object ActorMaterializer {
    * INTERNAL API
    */
   private[akka] def downcast(materializer: Materializer): ActorMaterializer =
-    materializer match {
+    materializer match { //FIXME this method is going to cause trouble for other Materializer implementations
```

Comment:
Yes exactly. But how is this method causing that? SynchronousFileSource is Actor based, so only Akka actor based materializers have a chance anyway, because it needs Akka features (hence the cast, which this method only makes more convenient and gives a uniform failure message). Custom materializers that want to support Akka actor based stages need to extend ActorMaterializer (which is not the same as ActorMaterializerImpl) which is public and abstract.